### PR TITLE
integration/container: use subtests for TestAttach

### PR DESCRIPTION
**Pull request by thaJeztah**
- Combine `TestAttachWithTTY` and `TestAttachWithoutTTy` to a single test using sub-tests
- Set up and tear-down the test-environment once
- Remove redundant client.ContainerRemove, as it's taken care of by testEnv.Clean()
- Run both tests in parallel

      make TEST_FILTER=TestAttach DOCKER_GRAPHDRIVER=overlay2 TESTDEBUG=1 test-integration
      Loaded image: busybox:latest
      Loaded image: busybox:glibc
      Loaded image: debian:bullseye-slim
      Loaded image: hello-world:latest
      Loaded image: arm32v7/hello-world:latest
      INFO: Testing against a local daemon
      === RUN   TestAttach
      === RUN   TestAttach/without_TTY
      === PAUSE TestAttach/without_TTY
      === RUN   TestAttach/with_TTY
      === PAUSE TestAttach/with_TTY
      === CONT  TestAttach/without_TTY
      === CONT  TestAttach/with_TTY
      --- PASS: TestAttach (0.00s)
          --- PASS: TestAttach/without_TTY (0.03s)
          --- PASS: TestAttach/with_TTY (0.03s)
      PASS

      DONE 3 tests in 1.347s

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like CI is broken, because the BuildKit tests don't pin to a minor Go version, so have been updated to go1.19.11;

```
2023-07-12T11:23:35.0244805Z #7 [internal] load metadata for docker.io/library/golang:1.19-alpine3.17
```


Which looks to have a regression;

- https://github.com/moby/moby/issues/45935

File: integration/container/attach_test.go [hunk 1]
```diff
@@ -10,41 +10,49 @@ import (
...
-	testAttach(t, true, types.MediaTypeRawStream)
-}
-
-func TestAttachWithoutTTy(t *testing.T) {
-	testAttach(t, false, types.MediaTypeMultiplexedStream)
-}
-
-func testAttach(t *testing.T, tty bool, expected string) {
+func TestAttach(t *testing.T) {
 	defer setupTest(t)()
```
[the remark this answers is not in the document]
Comment (vvoland) [reply integration/container/attach_test.go:14]: Hmmm actually it looks like we already have some tests which have this issue.

File: integration/container/attach_test.go [hunk 2]
```diff
@@ -10,41 +10,49 @@ import (
...
+			)
+			assert.NilError(t, err)
+			attach, err := client.ContainerAttach(context.Background(), resp.ID, types.ContainerAttachOptions{
+				Stdout: true,
+				Stderr: true,
+			})
+			assert.NilError(t, err)
+			mediaType, ok := attach.MediaType()
+			assert.Check(t, ok)
+			assert.Check(t, mediaType == tc.expectedMediaType)
```
Comment (vvoland) [inline integration/container/attach_test.go:55]:
```suggestion
			assert.Check(t, is.Equal(mediaType, tc.expectedMediaType))
```

**Review by vvoland — CHANGES_REQUESTED** [review]

File: integration/container/attach_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vvoland) [reply integration/container/attach_test.go:14]: Opened: https://github.com/moby/moby/pull/45957

[the remark this answers is not in the document]
Comment (thaJeztah) [reply integration/container/attach_test.go:14]: Doh! I have a branch somewhere that rewrites the setup test to use t.Cleanup, and I think I did it for this reason.

Let me go find it (or did I already open a PR for it? LOL)

File: integration/container/attach_test.go (hunk 2 above)
Comment (thaJeztah) [reply integration/container/attach_test.go]: Good catch; updated

**Review by vvoland — APPROVED** [review]

**Pull request merged**